pkg/guestagent/api/guestservice.proto

```diff
@@ -22,11 +24,20 @@ message Event {
 }
 
 message IPPort {
-  string ip = 1;
-  int32 port = 2;
+  string protocol = 1;
+  string ip = 2;
+  int32 port = 3;
```

Comment:
Perhaps this is a breaking change. According to https://protobuf.dev/programming-guides/proto3/#assigning  "This number cannot be changed once your message type is in use"

Better:

```suggestion
  string ip = 1;
  int32 port = 2;
  string protocol = 3;
```